### DISCUSSION TITLE
Export data to Excel

### DISCUSSION THREAD

**Discussion by VadimShadorin** [dmroeder/pylogix#243] [Q&A] [answered]
Hello Gents. Could You help with issue. 
Is there way to export whole massiv to column of excel spreadsheet
I have massiv getting from PLC using Pylogix - 190 elements. 
I can't find way to copy massiv to column.
Code 
------
import time
from pylogix import PLC
with PLC() as comm:
 comm.IPAddress = '192.168.1.104'
 read = True
 while read:
     try:
       ret = comm.Read('HMI_Analog_Array[0]',190)
       print(ret.Value)
       time.sleep(1)
     except KeyboardInterrupt:
       print('exiting')
       read = False

**Comment by evaldes2015** [comment]
Have you tried writing the data to a CSV file?
Have you looked at openpyxl?

**Reply by VadimShadorin** [reply]
Hello Evalsed2015. 
 Yes I did. I can do it only for one element from massiv but I can''t copy whole massive. I'm not an expert in Python unfortuantely. 
Let me look in openpyxl library description Maybe I missed smth.
Thanks

**Comment by evaldes2015** [comment, marked as answer]
Simple program to write to CSV:
from pylogix import PLC

plc = PLC()
plc.IPAddress = '192.168.128.22'
plc.ProcessorSlot = 0
data = plc.Read('conveyor', 300)
sdata = [str(d) for d in data.Value]
print(data)

f = open('data.csv', 'w')
f.writelines("\n".join(sdata))
f.close()

You'll obviously need to change the IP address, the tag name and the read length.

Good luck.

**Reply by VadimShadorin** [reply]
Thank You !
